fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/serialization/DefaultPlanSerializationRegistry.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * DefaultPlanSerializationRegistry.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.plan.serialization;
+
+import com.apple.foundationdb.record.PlanDeserializer;
+import com.apple.foundationdb.record.RecordCoreException;
+import com.apple.foundationdb.record.util.ServiceLoaderProvider;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Streams;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Message;
+import org.apache.commons.lang3.tuple.Pair;
+
+import javax.annotation.Nonnull;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Map;
+
+/**
+ * Default implementation of a plan serialization registry.
+ */
+public class DefaultPlanSerializationRegistry implements PlanSerializationRegistry {
+    public static final PlanSerializationRegistry INSTANCE = new DefaultPlanSerializationRegistry();
+    private static final String TYPE_URL_PREFIX = "c.a.fdb.types";
+
+    private final Map<Class<? extends Message>, PlanDeserializer<? extends Message, ?>> fromProtoClassDeserializerMap;
+    private final Map<String, Class<? extends Message>> fromProtoTypeUrlClassMap;
+
+    public DefaultPlanSerializationRegistry() {
+        final var methodMaps = loadFromProtoMethodMaps();
+        this.fromProtoClassDeserializerMap = methodMaps.getLeft();
+        this.fromProtoTypeUrlClassMap = methodMaps.getRight();
+    }
```

Comment:
It feels like this should be private to avoid people accidentally re-initializing this class (which it looks like we may be doing during tests)

Reply:
done